api/v1alpha1/author_webhook.go

```diff
@@ -0,0 +1,197 @@
+			return a.patchAuthorResponse(req.UserInfo.Username, &releasePlanNew)
+		}
+		return admission.Allowed("Success")
+	case admissionv1.Update:
+		var releasePlanOld ReleasePlan
+		err = json.Unmarshal(req.OldObject.Raw, &releasePlanOld)
+		if err != nil {
+			return admission.Errored(http.StatusInternalServerError, errors.Wrap(err, "error decoding object"))
+		}
+		if releasePlanNew.GetLabels()[metadata.AttributionLabel] != "true" {
+			// Attribution True -> False/Missing, remove author
+			if releasePlanOld.GetLabels()[metadata.AttributionLabel] == "true" {
+				return a.removeAuthorResponse(&releasePlanNew)
+			}
+			// Attribution False/Missing -> False/Missing, let request through
+			return admission.Allowed("Success")
+		}
+		author := releasePlanNew.GetLabels()[metadata.AuthorLabel]
+		// Attribution True -> True
+		if releasePlanOld.GetLabels()[metadata.AttributionLabel] == "true" {
+			// If changing author label to something beside current user, deny request
+			if author != "" && author != authorLabelValue(req.UserInfo.Username) &&
+				author != releasePlanOld.GetLabels()[metadata.AuthorLabel] {
+				return admission.Errored(http.StatusInternalServerError, errors.New("[labels] cannot change the author label if not setting"+
+					" it to the current user. Expected author is: "+authorLabelValue(req.UserInfo.Username)))
+			}
+			return admission.Allowed("Success")
+		}
+		if releasePlanNew.GetLabels()[metadata.AttributionLabel] == "true" &&
+			author != "" && author != authorLabelValue(req.UserInfo.Username) {
+			// Attribution False/Missing -> True, has request for different author, deny request
+			return admission.Errored(http.StatusInternalServerError, errors.New("[labels] cannot set attribution to true if not setting"+
+				" current user as author. Expected author is: "+authorLabelValue(req.UserInfo.Username)))
+		}
+		// Attribution False/Missing -> True, set author
+		return a.patchAuthorResponse(req.UserInfo.Username, &releasePlanNew)
+	default:
+		return admission.Allowed("Success")
+	}
+}
+
+// patchAuthorResponse returns an admission.Response that modifies the CR author label to be
+// the user provided.
+func (a *authorWebhook) patchAuthorResponse(username string, obj client.Object) admission.Response {
+	oldObj, err := json.Marshal(obj)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, errors.Wrap(err, "error encoding response object"))
+	}
+
+	objectLabels := make(map[string]string)
+	if obj.GetLabels() != nil {
+		objectLabels = obj.GetLabels()
+	}
+	objectLabels[metadata.AuthorLabel] = authorLabelValue(username)
+	obj.SetLabels(objectLabels)
+
+	newObj, err := json.Marshal(obj)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, errors.Wrap(err, "error encoding response object with author added"))
+	}
+
+	return admission.PatchResponseFromRaw(oldObj, newObj)
+}
+
+// removeAuthorResponse returns an admission.Response that removes the author label from the CR.
```

Comment:
I'd change it a bit. Main goal of the function is to remove the label. Returning a response is a side effect.

Reply:
You want an entire function just for `delete(obj.GetLabels(), metadata.AuthorLabel)`? The response is what removes it, as it returns a patch response